snapshot/bazel/filer_test.go

```diff
@@ -0,0 +1,189 @@
+	tempDir, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatalf("Error creating temp dir. %v", err)
+	}
+	_, err = noopBf.CheckoutAt("this definitely isn't right", tempDir.Dir)
+	if err == nil || !strings.Contains(err.Error(), invalidIdMsg) {
+		t.Fatalf("Expected checkout to be invalid due to ID")
+	}
+}
+
+func TestBzCheckouterValidCheckoutAt(t *testing.T) {
+	expectedSha := "01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b"
+	expectedSize := int64(5)
+	tempDir, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatalf("Error creating temp dir. %v", err)
+	}
+	snap, err := noopBf.CheckoutAt(fmt.Sprintf("bz-%v-%v", expectedSha, expectedSize), tempDir.Dir)
+	if err != nil {
+		t.Fatalf("Expected checkout to be valid. Err: %v", err)
+	}
+	if snap.ID() != expectedSha {
+		t.Fatalf("Expected snapshot ID to be %v, was %v", expectedSha, snap.ID())
+	}
+}
+
+// Ingester tests
+
+func TestBzIngesterGetFileTypeDir(t *testing.T) {
+	tmp, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatalf("Error creating temp dir. %v", err)
+	}
+
+	fileType, err := noopBf.getFileType(tmp.Dir)
+	if err != nil || fileType != directory {
+		t.Fatalf("Expected fileType to be %s, was %s. Err: %v", directory, fileType, err)
+	}
+
+	err = os.Remove(tmp.Dir)
+	if err != nil {
+		t.Fatalf("Error removing %v: %v", tmp.Dir, err)
+	}
+}
+
+func TestBzIngesterGetFileTypeFile(t *testing.T) {
+	tmp, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatalf("Error creating temp dir. %v", err)
+	}
+
+	tmpFile, err := tmp.TempFile("")
+	if err != nil {
+		t.Fatalf("Error creating temp file. %v", err)
+	}
+
+	fileType, err := noopBf.getFileType(tmpFile.Name())
+	if err != nil || fileType != file {
+		t.Fatalf("Expected fileType to be %s, was %s. Err: %v", file, fileType, err)
+	}
+
+	err = os.Remove(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("Error removing %v: %v", tmpFile.Name(), err)
+	}
+}
+
+func TestBzIngesterGetFileTypeInvalid(t *testing.T) {
+	_, err := noopBf.getFileType("some made up file or directory")
+	if err == nil || !strings.Contains(err.Error(), noSuchFileOrDirMsg) {
+		t.Fatalf("Expected error to contain %s, was %v", noSuchFileOrDirMsg, err)
+	}
+}
+
+func TestBzIngesterValidIngestDir(t *testing.T) {
+	tmp, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatalf("Error creating temp dir. %v", err)
+	}
+
+	_, err = noopBf.Ingest(tmp.Dir)
+	if err != nil {
+		t.Fatalf("Error ingesting dir %v. Err: %v", tmp.Dir, err)
+	}
+
+	err = os.Remove(tmp.Dir)
+	if err != nil {
+		t.Fatalf("Error removing %v: %v", tmp.Dir, err)
+	}
+}
+
+func TestBzIngesterValidIngestFile(t *testing.T) {
+	tmp, err := temp.TempDirDefault()
+	if err != nil {
+		t.Fatalf("Error creating temp dir. %v", err)
+	}
+
+	tmpFile, err := tmp.TempFile("")
+	if err != nil {
+		t.Fatalf("Error creating temp file. %v", err)
+	}
+
+	_, err = noopBf.Ingest(tmpFile.Name())
+	if err != nil {
+		t.Fatalf("Error ingesting file %v. Err: %v", tmpFile.Name(), err)
+	}
+
```

Comment:
would add a check against returned ID here to make sure it checks out - should be able to expect hardcoded hash and size 0 from empty temp file (see also later comment on ID returned form Ingest)